ocrd_tesserocr/segment_line.py

```diff
@@ -111,7 +111,19 @@ def process(self):
                         line_poly = Polygon(line_polygon)
                         if not line_poly.within(region_poly):
                             # this could happen due to rotation
-                            line_poly = line_poly.intersection(region_poly).convex_hull
+                            interline = line_poly.intersection(region_poly)
+                            if interline.is_empty:
+                                continue # ignore this line
+                            if hasattr(interline, 'geoms'):
+                                # is (heterogeneous) GeometryCollection
+                                area = 0
+                                for geom in interline.geoms:
+                                    if geom.area > area:
```

Comment:
This selects the largest geom in the collection, right? This is not a union, right?

Reply:
Exactly. Shapely returns a GeometryCollection when the result of the intersection is discontiguous (which we cannot represent in PAGE-XML coordinates). Thus, a union would only yield inconsistency w.r.t. the parent region again. Sometimes, the result can even be _heterogeneous_, i.e. line and point, polygon and line etc.

I thought looking at the area would be the safest/simplest way to handle all cases. But I have no good examples either way.